src/com/Communication.cpp

```diff
@@ -286,5 +294,70 @@ void Communication::broadcast(std::vector<double> &v, int rankBroadcaster)
   broadcast(v.data(), size, rankBroadcaster);
 }
 
+void Communication::writeConnectionInfo(std::string const & acceptorName,
+                                        std::string const & requesterName,
+                                        int rank,
+                                        std::string addressDirectory,
+                                        std::string addressData)
+{
+  using namespace boost::filesystem;
+  int const firstLevelLen = 2;
```

Comment:
```suggestion
  constexpr int firstLevelLen = 2;
```

This should probably be defined in the `Communication` class itself as a `static constexpr int`.

Reply:
Ok with `constexpr`, but I don't like making things `static` for no real reason.